src/v/kafka/server/handlers/fetch.cc

```diff
@@ -352,11 +351,16 @@ handle_shard_fetch(ss::shard_id shard, op_context& octx, shard_fetch fetch) {
         shard,
         octx.ssg,
         [foreign_read,
+         &octx,
          deadline = octx.deadline,
          configs = std::move(fetch.requests)](
           cluster::partition_manager& mgr) mutable {
             return fetch_ntps_in_parallel(
-              mgr, std::move(configs), foreign_read, deadline);
+              mgr,
+              octx.rctx.metadata_cache(),
```

Comment:
no because , confusingly enough, when you call the method `metadata_cache()` it calls `_cache.local()` , so this method returns the shard local reference.

Reply:
ahh right i forgot about that